### ISSUE TITLE
Add support for assisted manual tests

### ISSUE THREAD

**Issue by comps** [teemtee/tmt#1915]
It would be useful if TMT could automate certain steps of manual tests, and then let the user perform the manual part in an assisted way - either via some TMT-specific functionality (menu?) or just connecting stdio of the test to the user running `tmt` on the command line.

There is `tmt run ... --interactive`, but that seems like a debugging hack, ie. not capturing output of the automated part of the test, being outside of what plans can specify (configure), etc.  
I would imagine something like `plans/semi-manual.yaml`, which would tell `execute` to run in some assisted mode.

A typical use case would be `spec` file review.

1. `main.fmf` file would require `rpm` and `cpio`
2. `tmt` could be run with `-e ERRATUM=12345`
3. The test itself would `dnf download --source` a baseline version of the package
4. It would then contact ET and get a to-be-released SRPM of the same package
5. It would extract both via `rpm2cpio` and use `cpio` to output into a directory.

Finally, it would show a `select`-style menu (see `man bash`), giving the user the option to

1. See the new `.spec` file of the to-be-released package
2. Display a colored diff between the old/new versions
3. Submit an answer (is everything okay / is it not), which would dictate the test pass/fail state.
4. Clean everything up

So the user would basically just run a plan (on `local` or `1minutetip`) and then spend 10-30 seconds deciding, instead of spending 10 minutes of human time setting everything up and doing the work fully manually.

Some more use cases:

* Documentation review - spell check exceptions
* GUI testing - automated logic preparing everything up, human doing just the GUI part, ie. Libreoffice document signatures
* PDF/HTML visual sanity - that an OpenSCAP generated HTML report looks okay, table layout is not mangled, etc.
* Toolbox debugging - prepared environment (with automated setup) that lets the user interact with the SUT directly, ie. setting up libvirt, pre-installing some container, downloading and adding helper utilities to `PATH`, and giving interactive access to the user

**Comment by comps** [comment]
One big quality-of-life improvement would be preparing all the setup beforehand, and then requiring a single time block of human time to do all the verification, instead of waiting a long time between tests and their interactions.

Not sure how this would work as an implementation, maybe it's already solved by parametrizing the tests and doing two `execute` passes - first to download `require`s, set everything up, and a second to actually interact with the human.

**Comment by psss** [comment]
Yeah, something like this was discussed / brainstormed in early stages of the project: To allow manual steps to include parts which can be run automatically and then stop and allow user to do the (hopefully) small manual part. Adding to the backlog as a nice-to-have feature.
